.pre-commit-config.yaml

```diff
@@ -0,0 +1,37 @@
+exclude: "docs|node_modules|migrations|.git|.tox"
+default_stages: [commit]
+fail_fast: true
+
+repos:
+    - repo: https://github.com/pre-commit/pre-commit-hooks
+      rev: v3.4.0
+      hooks:
+          - id: trailing-whitespace
+          - id: end-of-file-fixer
+          - id: check-yaml
+
+    - repo: https://github.com/psf/black
+      rev: 21.9b0
+      hooks:
+          - id: black
+
+    - repo: https://github.com/timothycrosley/isort
+      rev: 5.9.3
+      hooks:
+          - id: isort
```

Comment:
`black` is a code formatter and `isort` is an import statement sorter.

I'm assuming the codebase did not integrate these QA tools previously. This means both of these tools need to be run on the codebase and perform formatting as needed. Otherwise, `pre-commit` will most likely throw error(s).

There are a few options moving forward
1) Run `pre-commit` locally and format everything.
  - This will produce a large diff
2) Exclude these for now and open up another issue if you want to use these QA tools 
3) Exclude them and don't use them (I personally prefer to use them)

Reply:
@tomvothecoder many thanks for the info. I think your guess is correct, that such tests have not been performed for the PMP codes. I think option 2 might be the  most efficient way to move forward.